### DISCUSSION TITLE
New Pomodoro Beta Testing

### DISCUSSION THREAD

**Discussion by Medvedev91** [Medvedev91/timeto.me#59] [General]
Current pomodoro works bad in real life, at least for me. So I tried many variations and found what I needed.

# Pomodoro Was Completely Remade

Before releasing a new version, I'd like to ask you to test that. I suggest testing it for 2 days and leave feedback.

The main feature is pressing the timer itself. This will pause the task. Pressing the timer again will restart the timer for paused task.

It means you only need one tap before the break and one tap after the break. As a pleasant side effect we have separated time tracking for breaks.

# Make a Backup

**Before installing the new apk BE SURE TO MAKE A BACKUP. Otherwise, you will lose data. The backup must be from the 535 version.**

# Download

[https://timeto.me/new_pomodoro_timetome.apk](https://timeto.me/new_pomodoro_timetome.apk) or join beta on Google Play (version 550+).

Looking forward to your feedback...

**Comment by ConEstArthur** [comment]
Great implementation. (At least for me)

It gives me new insights into using the app.
However
To start a new task, you will need to pause the current one and then start a new one and since **break** activity starts in between these two operations, your history will be filled with 2-3 seconds of **break** activity which to me is not a big deal but if you want your history to look neat, it is a nuisance.
(let me know if I'm missing something)

I cannot thank you enough for this update though. I do not have to go into any menus anymore. I'm a student and I've been using break.. study.. break.. study.. break study constantly for 2 days.  
Absolutely commendable work !!!

**Reply by Medvedev91** [reply]
> your history will be filled with 2-3 seconds of break activity

So far, I can't imagine a good solution. Let's leave it to the future.

**Reply by ConEstArthur** [reply]
I agree. 
Other features definitely take priority over this.

**Reply by ConEstArthur** [reply]
The only good solution I find for this is "Ignore Activities that are tracked under n seconds".

**Comment by motato1** [comment]
Lovely update!

The new break workflow is much more intuitive than before. No more breaks from breaks from breaks from breaks, when tapping the break button multiple times.

Some thoughts and things I've noticed:

1. I love the "Until Time" feature. So useful!
2. The `+` button increasing the timer in 5 minute increments is cool. However, being able to make the number negative, is not super useful imo. (Add 5min with `+` button, then reduce "Until Time"  by more than 5min)
1. Is there a need for the `i` button? Or is there a reason why the "Until Time" and "Timer" buttons are only visible when the timer is inversed (purple text)? As in, what's the connection between those elements?
1. This is mostly a nitpick, but the pause emoji/icon :pause_button: looks different in the home screen compared to the task screen:
![screenshot](https://github.com/Medvedev91/timetome/assets/70812484/8c2b73ac-71b4-4790-b8e3-c9c261a228b0)
3. The last thing is the behavior of the following workflow:
   * Start Activity A
   * Finish Activity A (timer has run out)
   * Start Break
   * Start Activity B
   
I now have an entry for a "paused" Activity A, even though I've finished it and moved on to do something else.

The behavior is not wrong per se, and an easy way to start your next cycle of Activity A is quite handy. But psychologically I would want to "tick all my boxes" to feel good about my day.

Not a big deal, I think. Just me being difficult ;)

Overall, I love the changes. Thank you so much for your effort @Medvedev91!

**Reply by ConEstArthur** [reply]
Great observations !!!
After this update I think readme will get even bigger :D

**Reply by Medvedev91** [reply]
> As in, what's the connection between those elements?

There's no point in showing these rare options all the time.

> This is mostly a nitpick, but the pause emoji/icon ⏸️ looks different in the home screen compared to the task screen:

This will be remade.

> I now have an entry for a "paused" Activity A, even though I've finished it and moved on to do something else.

The same. No ideas yet.

**Comment by ConEstArthur** [comment]
@Medvedev91 is it intentional that if I add a new task for Today. I cannot set it as "Important".
Nor can I set time for it. 
These options are only available for Recurring tasks.
(Other workaround is to just use calendar to add tasks but I think in future we should separate events from tasks)

**Reply by Medvedev91** [reply]
Right, task editing is one more thing that will be completely redesigned. :)

**Comment by motato1** [comment]
I have noticed one more thing that I find a bit confusing. Although, maybe that's intended behavior and my mental model doesn't match that.

Consider following workflow:
- Start Activity A for **1h**
- You get interrupted after 20 minutes and tap the timer to take a break from Activity A
  You have an entry for unfinished Activity A with **40 minutes** remaining
- Resume work by tapping the timer again
  The unfinished entry is removed and a new **1h timer** for Activity A is started

My expectation would be that the tapping the timer while on a break, resumes my last unfinished activity, not start a new timer for the same activity.

I know that I can just tap the paused entry and continue it, but, at least to me, it seems logical that the timer would acts as a toggle button between "current activity" and "break".

**Reply by Medvedev91** [reply]
Initial version worked like that. It seemed usable until I started testing it.

1. Sometimes I finish a task a little early, restarting the timer turns into a challenge;
2. Every time I resume, I can't be sure the timer will start with the same time;
3. Even if for some reason I want to restart the timer before the break, in the current implementation I can just double tap.

That's why I decided to restart the timer after resuming. But when I need to resume that option is also available, just like you said:

> I know that I can just tap the paused entry and continue it

---

I've made a lot of changes since the beta. I hope to release the final version next week.

**Reply by motato1** [reply]
That makes sense. Thank you for your answer (and your work :)).

I'm excited for the new release! :man_dancing:

**Reply by Medvedev91** [reply]
> I'm excited for the new release! 🕺

https://github.com/Medvedev91/timeto.me/releases/tag/v2024.07.08.560

Coming soon to all app stores.

**Reply by ConEstArthur** [reply]
> Coming soon to all app stores.

And we're here for it.

Rejoice, my lads! 😄

**Reply by motato1** [reply]
Fantastic  :rocket:  Thank you for the update!

**Comment by HighPriest** [comment]
I am once again asking for a persistent notification sound / alarm option, for tasks that have exceeded their assigned time.

Make it a global option or per category. Doesn't really matter. Just make something that can actually alert me.

I really never notice when the set time runs out and end up doing things waaay too long
